cache/lru_cache.cc

```diff
@@ -61,27 +65,26 @@ LRUHandle* LRUHandleTable::Remove(const Slice& key, uint32_t hash) {
 }
 
 LRUHandle** LRUHandleTable::FindPointer(const Slice& key, uint32_t hash) {
-  LRUHandle** ptr = &list_[hash & (length_ - 1)];
+  LRUHandle** ptr = &list_[hash >> (32 - length_bits_)];
   while (*ptr != nullptr && ((*ptr)->hash != hash || key != (*ptr)->key())) {
     ptr = &(*ptr)->next_hash;
   }
   return ptr;
 }
 
 void LRUHandleTable::Resize() {
-  uint32_t new_length = 16;
-  while (new_length < elems_ * 1.5) {
-    new_length *= 2;
-  }
-  LRUHandle** new_list = new LRUHandle*[new_length];
-  memset(new_list, 0, sizeof(new_list[0]) * new_length);
+  // FIXME: don't resize beyond shard_bits + length_bits == 32, unless
+  // we upgrade hash (storage) to 64 bits.
+  int old_length_bits = length_bits_;
+  int new_length_bits = length_bits_ + 1;
+  LRUHandle** new_list = new LRUHandle* [size_t{1} << new_length_bits] {};
   uint32_t count = 0;
-  for (uint32_t i = 0; i < length_; i++) {
+  for (uint32_t i = 0; (i >> old_length_bits) == 0; i++) {
```

Comment:
Is this more efficient than having a temporary variable "old_length = 1 << old_length_bits;"  and use old_length as your loop termination condition?